tap_s3_csv/sync.py

```diff
@@ -116,20 +119,26 @@ def handle_file(config, s3_path, table_spec, stream, extension, file_handler=Non
             # so we need to pass in fieldnames. First thread needs to handle first row if table_spec.has_header == True in order to avoid
             # having first row parsed as record when it's actually header. Set handle_first_row param for PreprocessStream to True
             # for this case so that the file/stream pointer is moved to skip first row.
+
+            # with import file copy for sftp, catalog is different from csv one, column_order is not present, so using cols_from_metadata
             file_handle = preprocess.PreprocessStream(
                 file_handle, table_spec, start_byte == 0 and table_spec.get('has_header', True))
-            fieldnames = stream['column_order']
+
+            fieldnames = cols_from_metadata
+
+            LOGGER.info(f'fieldnames: {fieldnames}')
+
         else:
             file_handle = s3.get_file_handle(config, s3_path)
-            if 'column_order' in stream:
+            if len(cols_from_metadata) > 0:
                 # same as above but for single thread. Set handle_first_row param to True if table_spec.has_header == True to avoid
                 # having header row parsed as first record
                 file_handle = preprocess.PreprocessStream(
                     file_handle, table_spec, table_spec.get('has_header', True))
-                fieldnames = stream['column_order']
+                fieldnames = cols_from_metadata
```

Comment:
column_order is only present in tap s3 csv config. We cannot depend on it as it will not be backwards compatible.